game-core/src/main/java/games/strategy/triplea/delegate/power/calculator/TotalPowerAndTotalRolls.java

```diff
@@ -13,6 +13,8 @@
 
   List<Die> getDiceHits(int[] dice);
 
+  int getDiceSides();
+
```

Comment:
Side-note, this interface I think has lost its cohesion, it is more than total power and total rolls. It's also odds as `TotalPowerAndTotalRolls` would seem to be the name for a value object rather than an interface. This is something that if we can would be good t come back and rename. I almost think this would be an even better interface for a 'unitGroup', or have that just be a class and get rid of the sub-classing altogether and have the `unitGroup` use this for an API that would 'just do the right thing'.

Reply:
This interface and `UnitPowerStrengthAndRolls` are two things I'm thinking of renaming/reworking for the casualty selection and support calculation.  I'm thinking of turning `UnitPowerStrengthAndRolls` (with a better name) into a delegate of `Unit`.  Then instead of passing around Units and calling `TotalPowerAndTotalRolls` to generate the `UnitPowerStrengthAndRolls` objects, the battle logic will first wrap all of the units in a `UnitPowerStrengthAndRolls` delegate and then those will be passed around.

I'm also planning on moving the dice calculation methods to `DiceRoll`.  The only reason they are on `TotalPowerAndTotalRolls` is legacy now.

Reply:
Cool. One note here, `DiceRoll` seems like it ought to be a value object, the number and types of methods on it do not seem appropriate. Something for you to consider, seemingly there should be of a strategy object that is more part of battles for determining roll results.